ledger/src/blockstore_meta.rs

```diff
@@ -637,12 +896,219 @@ mod test {
             .collect::<Vec<_>>()
             .choose_multiple(&mut rng, erasure_config.num_coding)
         {
-            index.coding_mut().index.remove(&idx);
+            index.coding_mut().remove(idx);
 
             assert!(!e_meta.should_recover_shreds(&index));
         }
     }
 
+    proptest! {
+        // Property: ShredIndexV2 can be converted to ShredIndex and vice versa
+        #[test]
+        fn shred_index_legacy_compat(
+            shreds in prop::collection::btree_set(
+                0..MAX_DATA_SHREDS_PER_SLOT,
+                0..MAX_DATA_SHREDS_PER_SLOT
+            ),
+            start in 0..MAX_DATA_SHREDS_PER_SLOT,
+            end in 0..MAX_DATA_SHREDS_PER_SLOT,
```

Comment:
I only skimmed the proptest crate docs so I could be misinformed, but is this generating test cases for all of these possibilities ? Ie, 32,768 possibilities ? How long does this test / the other tests take to run ?

Reply:
Ahh ok, so some randomness then ?

Reply:
@alessandrod raised a similar concern about test duration in below comment, so going to close this one and we can continue discussion there:
https://github.com/anza-xyz/agave/pull/3900/files#r1908160088